compiler/src/main/kotlin/deezer/kustomexport/compiler/js/mapping/TypeMapping.kt

```diff
@@ -45,9 +45,13 @@ class OriginTypeName(
     fun importedMethod(name: FormatString) = importMethod(name)
 
     val exportedTypeName: TypeName by lazy {
-        // Remove TypeParameter: because we can't export generic in a cool manner yet, so we produce concrete from generics.
-        // See @KustomExportGenerics
-        exportedType().removeTypeParameter()
+        var type = exportedType()
+        if(isKustomExportAnnotated) {
```

Comment:
@glureau to maintain backward compatibility with KustomGenerics I erased the type only if there is a kustom generic  specified for the given type

Reply:
Now the output is:
```
> Task :compiler:compileKotlin
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/GenericsVisitor.kt: (48, 28): Unchecked cast: Any? to List<KSAnnotation>
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/KspExt.kt: (52, 72): Unchecked cast: Any? to T
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/js/MethodNameDisambiguation.kt: (36, 68): 'capitalize(): String' is deprecated. Use replaceFirstChar instead.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/js/Structs.kt: (42, 5): Expected performance impact from inlining is insignificant. Inlining works best for functions with parameters of functional types
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/js/mapping/CustomMapping.kt: (184, 42): Parameter 'typeName' is never used, could be renamed to _
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/js/pattern/FunctionTransformer.kt: (31, 53): 'capitalize(): String' is deprecated. Use replaceFirstChar instead.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/compiler/src/main/kotlin/deezer/kustomexport/compiler/js/pattern/SealedClassTransformer.kt: (42, 9): Variable 'ctorParams' is never used

> Task :samples:compileKotlinJs
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/_class/sealed/js/ExtendedSealedParent.kt: (19, 11): 'when' is exhaustive so 'else' is redundant here
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/_class/sealed/js/SealedParent.kt: (26, 11): 'when' is exhaustive so 'else' is redundant here
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/Computer.kt: (34, 79): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (38, 68): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (45, 5): Exported declaration uses non-exportable return type: Promise<Unit>
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (46, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (52, 5): The expression is unused
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/IComputer.kt: (35, 79): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/IThingDoer.kt: (40, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/WebThingDoer.kt: (30, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/toplevelfunction/js/addsAsync.kt: (19, 21): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.

BUILD SUCCESSFUL in 5s
29 actionable tasks: 29 executed

> Task :samples:compileKotlinJs
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/_class/sealed/js/ExtendedSealedParent.kt: (19, 11): 'when' is exhaustive so 'else' is redundant here
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/_class/sealed/js/SealedParent.kt: (26, 11): 'when' is exhaustive so 'else' is redundant here
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/Computer.kt: (34, 79): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (38, 68): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (45, 5): Exported declaration uses non-exportable return type: Promise<Unit>
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (46, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/ComputerTester.kt: (52, 5): The expression is unused
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/IComputer.kt: (35, 79): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/IThingDoer.kt: (40, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/coroutines/js/WebThingDoer.kt: (30, 13): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.
w: /Users/MarcoSignoretto/Documents/careem/KustomExport/samples/build/generated/ksp/js/jsMain/kotlin/sample/toplevelfunction/js/addsAsync.kt: (19, 21): This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.

BUILD SUCCESSFUL in 1s
24 actionable tasks: 2 executed, 22 up-to-date

up to date, audited 19 packages in 518ms

found 0 vulnerabilities
✅   Generics: generics interface re-typed
✅   Generics: generics interface re-typed
✅   Generics: generics interface re-typed
✅   Generics: generics interface re-typed
✅   Exceptions: TS can interact with an object containing an Exception
✅   Exceptions: Kotlin can interface with an object containing an Exception
✅   Exceptions: Kotlin can retrieve the stacktrace from the exception
✅   MixingAnnotations: Retrieve data from a @JsExport class
✅   MixingAnnotations: Retrieve data from a @KustomExport class
✅   ValueClass: value class can wrap String
✅   ValueClass: value class can wrap Long
✅   ValueClass: value class can wrap Long
✅   CollectionMapping: mapping for List<Long>
✅   FunctionTypes: mapping for function types
✅   BasicTypeMapping: mapping for Boolean
✅   BasicTypeMapping: mapping for Byte
✅   BasicTypeMapping: mapping for Short
✅   BasicTypeMapping: mapping for Int
✅   BasicTypeMapping: mapping for Float
✅   BasicTypeMapping: mapping for Double
✅   BasicTypeMapping: mapping for String
//TODO: Typed arrays
✅   BasicTypeMapping: mapping for Any
✅   LongMapping: mapping for Long (2147483647)
✅   LongMapping: mapping for Long (4294967295)
✅   Enum: can consume enums in Kotlin code
✅   Enum: values() contains 4 items
✅   Enum: values() contains NORTH
✅   Enum: valueOf(NORTH) -> NORTH
✅   Enum: valueOf(UP) is null
✅   Enum: export doesn't duplicate wrappers
✅   Top-level function: is supported
✅   Top-level function: suspend is supported
✅   Coroutines: execute Kotlin coroutines
✅   Coroutines: execute Kotlin coroutines with 'then'
✅   Coroutines: parallel setTimeout
✅   Coroutines: reject throws an Error
✅   Coroutines: can cancel (timeout throw CancellationException)
✅   Coroutines: can cancel (work is aborted)
✅   Coroutines: can cancel Kotlin coroutines from Typescript (work has been cancelled and will never complete)
✅   InterfaceWithSuspend: executes suspend function with JS friendly types
✅   EmptyClass: can instantiate
✅   SealedClass: Sealed class 1
✅   SealedClass: Sealed class 2
✅   SealedClass: Extended sealed class
✅   SimpleClass: can retrieve value
✅   Inherit: can instantiate
✅   Cascade: direct child
✅   Cascade: sub childs
✅   Statics: static factory can be used
✅   DataClass: created from Kotlin and Typescript are equals
✅   Cascade: direct child
✅   Cascade: sub childs

```

Reply:
Fair enough, it'll handle collections from Kotlin as you need, and current behavior with annotated class is preserved. I presume tho that a kotlin.Result for example could be blocking, but that's ok from my point of view, generics export are really not great. (That's said, it may be better now, my last try was with Kt 1.6)